integration_test.go

```diff
@@ -3,51 +3,139 @@ package sockchat
 import (
 	"net/http/httptest"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
-func TestIntegration(t *testing.T) {
-	store, _ := NewSockChatStore()
+func TestAuthorizedUserFlow(t *testing.T) {
+	store, _ := NewChannelStore()
 	store.CreateChannel("foo")
-	store.CreateChannel("bar")
-	users := userStoreDouble{}
-	server := httptest.NewServer(NewSockChatServer(store, &users))
-	wsURL := GetWsURL(server.URL)
-
+	users := &userStoreDouble{}
+	server := httptest.NewServer(NewSockChatServer(store, users))
 	defer server.Close()
 
+	wsURL := GetWsURL(server.URL)
+
 	t.Run("user can send a message to a channel they have joined", func(t *testing.T) {
 		ws := NewTestWS(t, wsURL)
-		ws.Write(t, NewSocketMessage(LoginAction, LoginRequest{Nick: ValidUserNick, Password: ValidUserPassword}))
-		<-ws.MessageStash // discard the login response from server
 		defer ws.Close()
-
+		ws.Write(t, NewSocketMessage(LoginAction, LoginRequest{Nick: ValidUserNick, Password: ValidUserPassword}))
 		ws.Write(t, NewSocketMessage(JoinAction, ChannelRequest{"foo"}))
-		<-ws.MessageStash //discard the `channel_joined` from server
-
 		ws.Write(t, NewSocketMessage(SendMessageAction, SendMessageRequest{Channel: "foo", Text: "hi!"}))
 
-		received := <-ws.MessageStash
-		wantAction := "new_message"
-		require.Equal(t, wantAction, received.Action, "action should be %s", wantAction)
-		message := UnmarshalMessageEvent(received.Payload)
-		wantAuthor := ValidUserNick
-		assert.Equal(t, wantAuthor, message.Author, "author should be %s", wantAuthor)
-
+		ws.AssertEventReceivedWithin(t, NewMessageEvent, 100*time.Millisecond)
```

Comment:
This test fails for me when I'm running tests with race detector `go test -race -count=1 ./...`:
```
--- FAIL: TestAuthorizedUserFlow (0.10s)
    --- FAIL: TestAuthorizedUserFlow/user_can_send_a_message_to_a_channel_they_have_joined (0.10s)
        integration_test.go:26: assertion failed - timed out waiting for websocket event: new message in channel
panic: repeated read on failed websocket connection
```
Consider increasing it here and in other places (2 seconds work for me after applying all the other fixes)